cloudkittydashboard/dashboards/project/reporting/views.py

```diff
@@ -31,66 +37,156 @@ def _do_this_month(data):
-            hourly_data = service_data['hourly']
+                service_id, {"cumulated": 0, "hourly": {}}
+            )
+            service_data["cumulated"] += decimal.Decimal(resource["rating"])
+            hourly_data = service_data["hourly"]
             hourly_data.setdefault(timestamp, 0)
-            hourly_data[timestamp] += float(resource['rating'])
+            hourly_data[timestamp] += float(resource["rating"])
 
     service_names = services.keys()
     t = start_timestamp
     if end_timestamp:
         while t <= end_timestamp:
             for service in service_names:
-                hourly_d = services[service]['hourly']
+                hourly_d = services[service]["hourly"]
                 hourly_d.setdefault(t, 0)
             t += 3600
 
     # now sort the dicts
     for service in service_names:
-        d = services[service]['hourly']
-        services[service]['hourly'] = collections.OrderedDict(
-            sorted(d.items(), key=lambda t: t[0]))
+        d = services[service]["hourly"]
+        services[service]["hourly"] = collections.OrderedDict(
+            sorted(d.items(), key=lambda t: t[0])
+        )
 
     return services
 
 
 class CostRepartitionTab(tabs.Tab):
     name = "This month"
     slug = "this_month"
-    template_name = 'project/reporting/this_month.html'
+    template_name = "project/reporting/this_month.html"
 
     def get_context_data(self, request, **kwargs):
         today = datetime.datetime.today()
         day_start, day_end = calendar.monthrange(today.year, today.month)
-        begin = "%4d-%02d-01T00:00:00" % (today.year, today.month)
-        end = "%4d-%02d-%02dT23:59:59" % (today.year, today.month, day_end)
+
+        form = self.get_form()
+
+        if form.is_valid():
+            # set values to be from datepicker form
+            start = form.cleaned_data["start"]
+            end = form.cleaned_data["end"]
+            begin = "%4d-%02d-%02dT00:00:00" % (start.year,
+                                                start.month, start.day)
+            end = "%4d-%02d-%02dT23:59:59" % (end.year, end.month, end.day)
+
+            if end < begin:
+                messages.error(
+                    self.request,
+                    _(
+                        "Invalid time period. The end date should be "
+                        "more recent than the start date."
+                        " Setting the end as today."
+                    ),
+                )
+
+                end = "%4d-%02d-%02dT23:59:59" % (today.year,
+                                                  today.month, day_end)
+
+            elif start > today.date():
+                messages.error(
+                    self.request,
+                    _(
+                        "Invalid time period. You are requesting "
+                        "data from the future which may not exist."
+                    ),
+                )
+
+        elif form.is_bound:  # when the form is invalid
```

Comment:
Don't think you need this comment
```suggestion
        elif form.is_bound:
```